### PR TITLE
screenName

### PR DESCRIPTION
**Pull request by Oleksii-Pavliuk**
QuerryPrep function added  which prepares queries before executing them. So it makes it run extremely fast on my opinion and will read data one time for each combination

### CODE REVIEW

File: functions/src/screenName.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+export const ScreenName = functions.https.onRequest(async (req, res) => {
+	//getting variables from request and checking them
+	if (req.method == "GET") {
+		var name: any = req.query.name as string;
+		var precision: any = req.query.precision;
+	} else if (req.method == "POST") {
+		var name: any = req.body.name as string;
+		var precision = req.body.precision;
+	}
+	if (!name|| name.length < 7) {
```
Comment (kovarny) [inline functions/src/screenName.ts:14]: let's use a constant instead of number

File: functions/src/screenName.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		res.status(400).send('Parameter "name" - min 7 characters');
+		return;
+	}else if (!precision || precision < 0.9 || precision > 1) {
+		res.status(400).send('Parameter 0.9 < "precison" < 1');
+		return;
+	}
```
Comment (kovarny) [inline functions/src/screenName.ts:20]: Let's remove the limit of precision now that we have the limit in place.

**Review by kovarny — CHANGES_REQUESTED** [review]

File: functions/src/screenName.ts (hunk 1 above)
Comment (Oleksii-Pavliuk) [reply functions/src/screenName.ts]: Const added

File: functions/src/screenName.ts (hunk 2 above)
Comment (Oleksii-Pavliuk) [reply functions/src/screenName.ts]: No limit no more

**Review by kovarny — APPROVED** [review]

**Pull request merged**